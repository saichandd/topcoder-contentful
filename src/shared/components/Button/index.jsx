/**
 * Button component which accepts via props the text to render
 * inside the button, its size and style.
 *
 * On hover, mouse cursor should become pointer, and the onClick()
 * callback passed from parent should be triggered on click.
 *
 * Style:
 * To create outline btn pass 'tc-outline-btn' className
 * To create filled btn pass '{color} tc-blue-btn' className. eg. 'red tc-blue-btn'
 * To create big btn pass 'tc-bg-btn' className along with one of the above two btn type
 */

/* TODO: In its current form this component is not that userful. Should we
 * get rid of it, just replacing with direct use of <button> tag? */

import React from 'react';
import PT from 'prop-types';
import _ from 'lodash';
import './styles.scss';

export default function Button({
  children,
  className,
  onClick,
}) {
  return (
    <button
      className={`tc-btn ${className}`}
      onClick={() => onClick()}
      type="button"
    >
      {children}
    </button>
  );
}

Button.defaultProps = {
  className: 'tc-blue-btn',
  onClick: _.noop,
  children: null,
};

Button.propTypes = {
  className: PT.string,
  onClick: PT.func,
  children: PT.node,
};
