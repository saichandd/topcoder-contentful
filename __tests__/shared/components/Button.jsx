import Button from 'components/Button';
import React from 'react';
import TU from 'react-dom/test-utils';
import { JU } from 'topcoder-react-utils';

test('Snapshot match', () => {
  JU.snapshot(<Button />);
});

class ButtonClass extends React.Component {
  componentDidMount() {}

  render() {
    return <Button {...this.props} />;
  }
}

const mockOnClick = jest.fn();

const render = TU.renderIntoDocument((
  <ButtonClass
    onClick={mockOnClick}
  />
));

test('onClick', () => {
  const obj = TU.findRenderedDOMComponentWithClass(render, 'tc-btn');
  TU.Simulate.click(obj);
  expect(mockOnClick).toHaveBeenCalled();
});
