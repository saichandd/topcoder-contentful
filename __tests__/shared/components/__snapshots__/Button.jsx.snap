// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot match 1`] = `
<button
  className="tc-btn tc-blue-btn"
  onClick={[Function]}
  type="button"
/>
`;
